### DISCUSSION TITLE
Temporarily set `loot-tables` to `false` whilst `MC-218156`

### DISCUSSION THREAD

**Discussion by 42willow** [PaperMC/Paper#10536] [Ideas]
### Is your feature request related to a problem?

Yes, https://bugs.mojang.com/browse/MC-218156
Context: My world is completely pre generated though, this issue occurs 100% of the time because of this

### Describe the solution you'd like.

Temporarily change to the default configuration whilst https://bugs.mojang.com/browse/MC-218156 isn't patched

```diff
  treasure-maps:
    enabled: true
    find-already-discovered:
-      loot-tables: 'default'
+      loot-tables: 'false'
      villager-trade: false
```

### Describe alternatives you've considered.

Wait for https://bugs.mojang.com/browse/MC-218156 to fix itself

### Other

Here is what happens on my server (as far as I know)

The buried treasure map is generated by the server searching to find an unexplored/ungenerated buried treasure. This doesn't work on my server, as all chunks are already pre generated / "explored" (by Chunky)

To fix this issue, I disabled the check the server does to see if the chest has already been looted (chunk generated). This was successful.

**Comment by Machine-Maker** [comment]
I don't think we should change the default behavior because some servers go through the process of pregenerating the terrain. Pre-gen isn't a vanilla or paper thing, its a third-party thing. 

We already provide an option to change the setting as you've demonstrated. Is there a reason that people who pregen can't just change that setting themselves?

**Comment by 42willow** [comment]
No there isn't a reason why people who pregen can't change the setting themselves, it just isn't well documented, and a bit of a hassle to figure out the cause and restart to apply changes. 

Alternatively, there should be some form of error message sent to the user and a verbose error in the console.

The behaviour change also isn't that drastic and (in my opinion) it's smarter not to ignore generated chunks. Even if a pregen wasn't run, a smallish server would still have the issue if the area is thoroughly explored and the shipwreck chests aren't sought out immediately.

**Comment by 42willow** [comment]
It's been two years, and the bug is still unresolved - I don't think it is going to be fixed any time soon, as Mojang doesn't seem to be that fast at patching things..

**Comment by Machine-Maker** [comment]
I wouldn't be against some log message when the server tries to generate a map but fails to find a valid location. We can check the state of that config message and log out a suggestion to toggle that setting. But I don't think we should change the default for the setting.

**Comment by Chris6ix** [comment]
> Mojang doesn't seem to be that fast at patching things..

That is quite the understatement lmao. 2 years is nothing. I'm still waiting for a bug to be fixed that will soon get it's 10-year anniversary. ([MC-53602](https://bugs.mojang.com/browse/MC-53602)).
Being a Mojang dev must be literally the easiest job in the world.

**Comment by 42willow** [comment]
Maybe it would be better if this was just updated in the pregen's documentation then. Hopefully if someone encounters this problem they will come across this discussion. Hopefully it gets patched eventually, but it really isn't that big of a deal in the grand scheme of things...

https://github.com/pop4959/Chunky/issues/336
